tests/evaluate/test_default_evaluator_delta.py

```diff
@@ -0,0 +1,142 @@
+import shutil
+import tempfile
+
+import pandas as pd
+import pytest
+from pyspark.sql import SparkSession
+
+import mlflow
+from mlflow.exceptions import MlflowException
+
+
+def language_model(inputs: list[str]) -> list[str]:
+    return inputs
+
+
+def test_write_to_delta_fails_without_spark():
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            artifact_path="model", python_model=language_model, input_example=["a", "b"]
+        )
+        data = pd.DataFrame({"text": ["Hello world", "My name is MLflow"]})
+        with pytest.raises(
+            MlflowException,
+            match="eval_results_path is only supported in Spark environment",
+        ):
+            mlflow.evaluate(
+                model_info.model_uri,
+                data,
+                extra_metrics=[mlflow.metrics.latency()],
+                evaluators="default",
+                evaluator_config={
+                    "eval_results_path": "my_path",
+                    "eval_results_mode": "overwrite",
+                },
+            )
+
+
+@pytest.fixture
+def spark_session_with_tempdir():
+    tmpdir = tempfile.mkdtemp()
+    spark = (
+        SparkSession.builder.master("local[*]")
+        .config("spark.jars.packages", "io.delta:delta-spark_2.12:3.0.0")
+        .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+        .config(
+            "spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog"
+        )
+        .config("spark.sql.warehouse.dir", tmpdir)
+        .getOrCreate()
+    )
+
+    yield spark, tmpdir
+
+    spark.stop()
+    shutil.rmtree(tmpdir)
```

Comment:
Could we use context manager instead?
```suggestion
    with tempfile.TemporaryDirectory() as tmpdir, (
        SparkSession.builder.master("local[*]")
        .config("spark.jars.packages", "io.delta:delta-spark_2.12:3.0.0")
        .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
        .config(
            "spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog"
        )
        .config("spark.sql.warehouse.dir", tmpdir)
        .getOrCreate() as spark:
            yield spark, tmpdir
```